pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,177 @@
+package resourcerecordset
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	//RRSetNotFound is the error code that is returned if RRSet is present
+	RRSetNotFound = "InvalidRRSetName.NotFound"
+)
+
+// Client defines ResourceRecordSet operations
+type Client interface {
+	ChangeResourceRecordSetsRequest(input *route53.ChangeResourceRecordSetsInput) route53.ChangeResourceRecordSetsRequest
+	ListResourceRecordSetsRequest(input *route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest
+}
+
+// NotFound will be raised when there is no ResourceRecordSet
+type NotFound struct{}
+
+func (err *NotFound) Error() string {
+	return fmt.Sprint(RRSetNotFound)
+}
+
+// NewClient creates new AWS client with provided AWS Configuration/Credentials
+func NewClient(config *aws.Config) Client {
+	return route53.New(*config)
+}
+
+// GenerateChangeResourceRecordSetsInput prepares input for a ChangeResourceRecordSetsInput
+func GenerateChangeResourceRecordSetsInput(p *v1alpha3.ResourceRecordSetParameters, action route53.ChangeAction) *route53.ChangeResourceRecordSetsInput {
+	var ttl *int64
+	if p.TTL == nil {
+		num := int64(300)
+		ttl = &num
+	} else {
+		ttl = p.TTL
+	}
+
+	resourceRecords := make([]route53.ResourceRecord, 0)
```

Comment:
```suggestion
	resourceRecords := make([]route53.ResourceRecord, 0, len(p.Records))
```

It seems like we always know how many records there will be, so we can be a little more efficient by pre-allocating a backing array of that length for our slice.

Reply:
That makes sense, I have added `len(p.Records)` to the make call.